Sample readme final review (#90)
OfficeDev/Office-Samples: Excel-Add-in-Custom-Function-Using-WebWorker/README.md

- Office connected to a Microsoft 365 subscription. You might qualify for a Microsoft 365 E5 developer subscription through the [Microsoft 365 Developer Program](https://developer.microsoft.com/microsoft-365/dev-program), see [FAQ](https://learn.microsoft.com/office/developer-program/microsoft-365-developer-program-faq#who-qualifies-for-a-microsoft-365-e5-developer-subscription-) for details. Alternatively, you can [sign up for a 1-month free trial](https://www.microsoft.com/microsoft-365/try?rtc=1) or [purchase a Microsoft 365 plan](https://www.microsoft.com/microsoft-365/buy/compare-all-microsoft-365-products).
  
### Run the add-in using Office Add-ins Development Kit extension

We recommend you try this sample by using the [Office Add-ins Development Kit](https://marketplace.visualstudio.com/items?itemName=msoffice.microsoft-office-add-in-debugger). The Office Add-ins Development Kit is an end-to-end developer tool for building Office add-ins. It helps create, run, and debug an Office Add-in.

1. **Download the sample code**

>    You can use either of the ways to download the sample code:

Reviewer: Grammar suggestion, here and elsewhere:
```suggestion
   To download this sample code, either:
```

   * Open the Office Add-ins Development Kit extension and view samples in the **Sample gallery**. Select the **Create** button in the top-right corner of the sample page.
   * [Clone](https://docs.github.com/repositories/creating-and-managing-repositories/cloning-a-repository) this repository or download this sample to a folder on your computer. Then, open the folder in Visual Studio Code.
   
1. **Open the Office Add-ins Development Kit**
    
    Select the <img src="./assets/Icon_Office_Add-ins_Development_Kit.png" width="30" alt="Office Add-ins Development Kit"/> icon in the **Activity Bar** to open the extension.

1. **Preview Your Office Add-in (F5)**

>     Select **Preview Your Office Add-in(F5)** to launch the add-in and debug the code. In the drop down menu, select the option **Desktop (Edge Chromium)**.

Reviewer: The option seems to be called "Edge Desktop (Edge Chromium)", so we should be consistent here. Also, let's use the proper term for the dropdown menu, "Quick Pick".
```suggestion
    Select **Preview Your Office Add-in(F5)** to launch the add-in and debug the code. In the Quick Pick menu, select the option **Edge Desktop (Edge Chromium)**.
```

>     Select **Preview Your Office Add-in(F5)** to launch the add-in and debug the code. In the drop down menu, select the option **Desktop (Edge Chromium)**.

Reviewer: According to the MSFT style guide, we should avoid referring to UI controls by their type (i.e.: "dropdown menu"). https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28800

What is this menu actually called in VS Code?

Reviewer 2: It's called "[Quick Pick](https://code.visualstudio.com/api/ux-guidelines/overview#quick-pick)"

>     <img src="./assets/devkit_preview.png" width="500" alt="Screenshot shows Preview your Office add-in in Office Add-ins Development Kit"/>

Reviewer: A few issues with this screenshot:
1. The alt text should not describe the image as a "screenshot", but rather what the screenshot is of. It's redundant because the screen reader already describes this as an image. 
```suggestion
    <img src="./assets/devkit_preview.png" width="500" alt="The Office Add-ins Development Kit task pane options to build and test your add-in."/>
```
2. The image has a lot of text. This makes the alt text hard to write because you'd need to include all that text in the alt text (if someone can't see, they still need the information from the image).
3. Additionally, images with text can't be translated. 
4. I find the image's visual flow to be off. Everything is in English, so I'm reading top-to-bottom, left-to-right. The first thing I see is the red box with the menu options, but I think you intend the Dev Kit icon to be the first thing readers see (since they'll need to click there first).
<img width="711" alt="image" src="https://github.com/user-attachments/assets/51fdea94-d65e-4698-9888-81902ff5ee41">

Reviewer 2: I think this image is at stale and should be replaced. I can't find it in the pr files, @hermanwenhe to confirm we have deleted it.

|   | launch.json               Launch and debug configurations
|   | settings.json             
|   | tasks.json                
| assets/                       Static assets, such as images
| babel.config.json
| manifest*.xml                 Manifest file
| package.json                  
| README.md                     
| RUN_WITH_EXTENSION.md         
| SECURITY.md
| src/                          Add-ins source code
|   | commands/
|   |   | commands.html
|   |   | commands.js
|   | taskpane/

> |   |   | taskpane.html         Task pane entry html

Reviewer: Outside of variable or file names, use the proper capitalization for terms like HTML.
```suggestion
|   |   | taskpane.html         Task pane entry HTML
```

|   |   | taskpane.js           Add API calls and logic here
|   | functions/

> |   |   | functions.js          Custom function js
> |   |   | functions-worker.js   Web worker js

Reviewer: Similar suggestion as above.
```suggestion
|   |   | functions.js          Custom function JavaScript
|   |   | functions-worker.js   Web worker JavaScript
```

If you edit the manifest as part of your changes, use the **Validate Manifest File** option in the Office Add-ins Development Kit. This shows you errors in the manifest syntax.

## Engage with the team

Did you experience any problems with the sample? [Create an issue]( https://github.com/OfficeDev/Office-Samples/issues/new) and we'll help you out.

Want to learn more about new features and best practices for the Office platform? [Join the Microsoft Office Add-ins community call](https://learn.microsoft.com/office/dev/add-ins/overview/office-add-ins-community-call).

## Copyright

Copyright (c) 2024 Microsoft Corporation. All rights reserved.
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information, see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
<br>**Note**: The taskpane.html file contains an image URL that tracks diagnostic data for this sample add-in. Please remove the image tag if you reuse this sample in your own code project.

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Author: This pixel image is not actually added in all sample task panes. Adding these means extra work and hight maintenance effort, we should carefully discuss the necessity of this.

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Author: I've restored this statement of those actually have this 1-pixel image in the taskpane.html.

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Reviewer: Please work with @davidchesnut here, as he set this up for the main samples repo and shares the data with the broader team.

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Author: @MingjiaLiu1995 Can you follow up on this? I think the first question mark is whether this is a blocker for public preview.

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Reviewer 2: @hermanwenhe Can you be more specific on what kind of extra work we need to add the pixel image? It seems the image is already hosted and we need to align the naming principle with David?

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Author: @MingjiaLiu1995 I might misunderstand something. If your goal is to measure the traffic to GitHub repo, then it can't be achieved. This is because the readme content will be reused in sample gallery so the same 1 pixel image will be loaded in the detail page of sample gallery.

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Reviewer 2: Synced offline. We will align the telemetry with other samples to indicate the popularity.

> <img src="https://pnptelemetry.azurewebsites.net/pnp-officeaddins/samples/word-add-in-aigc">

Author: I will use another PR to address this.
